Add fp32 to in8 conversion notebook. Removed imread-from-url library dependency. Minor fixes

mokshmalik5757 opened the pull request:
I have made some minor fixes in your code. Changed .gitignore file extension. Removed imread-from-url dependency and created a custom function using requests library for reading images from URL. Fixed the requirements.txt. Changed function in sam2/utils.py.

mokshmalik5757 committed:
Add fp32 to in8 conversion notebook. Removed imread-from-url library dependency. Minor fixes

mokshmalik5757 commented:
I have made some minor fixes in your code. Changed .gitignore file extension. Removed imread-from-url dependency and created a custom function using requests library for reading images from URL. Fixed the requirements.txt. Changed function in sam2/utils.py.

Commit:
Change sam2. Add read_from_url func.

ibaiGorordo commented:
In my case I get Error 403, when trying to get the image. The error disappears using this instead:
```
        ua = UserAgent()
        header = {'User-Agent': str(ua.chrome)}
        response = requests.get(img_url, headers=header)

mokshmalik5757 committed:
Add imread_from_url back. Removed custom function

mokshmalik5757 committed:
Merge branch 'ONNX-fp32-to-int8' of https://github.com/mokshmalik5757/ONNX-SAM2-Segment-Anything into ONNX-fp32-to-int8

mokshmalik5757 commented:
I have reverted back the changes with "imread_from_url" dependency, and I have also fixed the error `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()` while executing the `if image` check.

mokshmalik5757 commented:
I'd also like to suggest that you should add the notebook for converting the PyTorch model of SAM 2 to ONNX in the notebooks directory that I've created, that way, one would be able to reference all the conversion code in a single directory.

mokshmalik5757 committed:
Merge branch 'main' into ONNX-fp32-to-int8
